code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -1236,24 +1251,29 @@ pub mod pallet {
 			let mut total_weight = unix_time_now_weight;
 
 			RewardPools::<T>::translate(|pool_id, mut reward_pool: RewardPoolOf<T>| {
-				for (asset_id, reward) in &mut reward_pool.rewards {
-					Self::reward_accumulation_hook_reward_update_calculation(
-						pool_id,
-						*asset_id,
-						reward,
-						now_seconds,
-					);
-				}
+				// If reward pool has not started, do not accumulate rewards or adjust weight
+				if reward_pool.start_block <= frame_system::Pallet::<T>::current_block_number() {
+					for (asset_id, reward) in &mut reward_pool.rewards {
+						Self::reward_accumulation_hook_reward_update_calculation(
+							pool_id,
+							*asset_id,
+							reward,
+							now_seconds,
+						);
+					}
 
-				// reward_pool.rewards is limited T::MaxRewardConfigsPerPool, which is Get<u32>
-				let number_of_rewards_in_pool = reward_pool.rewards.len() as u64;
+					// reward_pool.rewards is limited T::MaxRewardConfigsPerPool, which is Get<u32>
+					let number_of_rewards_in_pool = reward_pool.rewards.len() as u64;
 
-				total_weight += (number_of_rewards_in_pool * T::WeightInfo::reward_accumulation_hook_reward_update_calculation()) +
+					total_weight += (number_of_rewards_in_pool * T::WeightInfo::reward_accumulation_hook_reward_update_calculation()) +
 					// NOTE: `StorageMap::translate` does one read and one write per item
 					T::DbWeight::get().reads(1) +
 					T::DbWeight::get().writes(1);
 
-				Some(reward_pool)
+					Some(reward_pool)
+				} else {
```

Comment:
I guess we don't need the else here at all

Reply:
71bff2c170a362c8f1b75f9f5d6b2f846e4836d5